### PR TITLE
Remove start, end and expiration from all licenses

### LINKED ISSUE #1116: Only use license table expiration
We have start and end dates for resource and workflow licenses stored separately and in the common license table. We should have only one place for the, i.e. the common table.

Now a license can be expired in the resource licenses sense but not in the general license sense. And the API will happily return it when you get all licenses.

- [ ] only have start and end dates in licenses table
- [ ] migration for old db
- [ ] test for this case of expiration in the API

### PR DESCRIPTION
**Pull request by opqdonut**
Fixes #1116 

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## API
- [ ] API is backwards compatible or completely new

## Documentation
- [x] update changelog if necessary

## Testing
- [x] complex logic is unit tested
- [x] valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] new tasks are created for pending or remaining tasks
- [x] no critical TODOs left to implement

### CODE REVIEW

File: test/clj/rems/api/test_workflows.clj [hunk 1]
```diff
@@ -99,10 +99,9 @@
                              :handlers [{:userid "bob" :email "bob@example.com" :name "Bob Approver"}
                                         {:userid "carl" :email "carl@example.com" :name "Carl Reviewer"}]}
                   :enabled true
-                  :expired false
                   :archived false}
         fetch (fn []
-                (let [wfs (-> (request :get "/api/workflows" {:archived true :expired true :disabled true})
+                (let [wfs (-> (request :get "/api/workflows" {:archived true :disabled true})
```
Comment (luontola) [inline test/clj/rems/api/test_workflows.clj]: The workflow API still has expired.

Comment (opqdonut) [reply test/clj/rems/api/test_workflows.clj]: bah I botched a revert. I'll amend

File: test/clj/rems/db/test_licenses.clj [hunk 2]
```diff
@@ -24,23 +24,15 @@
...
+                                            {:id :normal2 :enabled true :archived false}
+                                            {:id :disabled :enabled false :archived false}
+                                            {:id :archived :enabled true :archived true}])]
+    (testing "filters"
+      (is (= [:normal :normal2 :disabled :archived]
+             (map :id (get-all-licenses {}))))
+      (is (= [:archived]
+             (map :id (get-all-licenses {:archived true}))))
+      (is (= [:normal :normal2]
+             (map :id (get-all-licenses {:archived false :enabled true})))))))
```
Comment (luontola) [inline test/clj/rems/db/test_licenses.clj]: This test is missing come cases: `{:archived false}`, `{:enabled true}`, `{:enabled false}`

Possibly also the various combinations of :archived and :enabled, if they are interesting.

Comment (opqdonut) [reply test/clj/rems/db/test_licenses.clj]: I'll write more tests

**Pull request merged**